libs/core/src/main/java/org/opensearch/core/common/io/stream/BaseWriteable.java

```diff
@@ -48,10 +48,15 @@ class WriteableRegistry {
          * @opensearch.internal
          */
         public static <R extends Reader<? extends BaseStreamInput, ?>> void registerReader(final byte ordinal, final R reader) {
-            if (READER_REGISTRY.containsKey(ordinal)) {
+            if (READER_REGISTRY.putIfAbsent(ordinal, reader) != null) {
                 throw new IllegalArgumentException("Streamable reader already registered for ordinal [" + (int) ordinal + "]");
             }
-            READER_REGISTRY.put(ordinal, reader);
+        }
+
+        public static void registerWriterCustomClass(final Class<?> classInstance, final Class<?> classGeneric) {
```

Comment:
Mind if I suggest to rename this method to `registerClassAlias` / `registerClassMap` (or something along this lines): when I first read the code, I expected to see custom `Writer<?>` (since it says `registerWriterXxx`), but as you explained, we just have some sort of alias for a class, not a writer. Does it make sense?

Reply:
Sure thing! I agree it makes it easier to follow.